src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,116 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
 
-export const NewMovie = () => {
+interface Props {
+  onAdd: (
+    movie: {
+      title: string,
+      description: string,
+      imgUrl: string,
+      imdbUrl: string,
+      imdbId: string,
+    }
+  ) => void
+}
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+  const [newMovie, setNewMovie] = useState({
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
```

Comment:
move initial state to a variable `initialMovieState` (outside the component)
```suggestion
  const [newMovie, setNewMovie] = useState<Movie>(initialMovieState);
```